crates/sui/src/sui_commands.rs

```diff
@@ -627,6 +641,13 @@ async fn start(
             !no_full_node,
             "Cannot start the indexer without a fullnode."
         );
+        // the indexer requires to set the fullnode's data ingestion directory
+        // note that this overrides the default configuration that is set when running the genesis
+        // command, which sets data_ingestion_dir to None.
+        ensure!(
+            with_data_ingestion.is_some(),
+            "Cannot start the indexer without --with-data-ingestion."
+        );
```

Comment:
We can do something like this instead to retain the backwards compatibility:

```suggestion
    }
    
    if with_indexer.is_some() && with_data_ingestion.is_none() {
        with_data_ingestion = Some(tempdir()?.into_path())
    }
```